ecosystem/nft-metadata-crawler/nft-metadata-crawler-parser/src/models/nft_metadata_crawler_uris_query.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright © Aptos Foundation
+
+use crate::schema::nft_metadata_crawler_uris;
+use diesel::{
+    prelude::*,
+    r2d2::{ConnectionManager, PooledConnection},
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Deserialize, Identifiable, Queryable, Serialize)]
+#[diesel(primary_key(token_uri))]
+#[diesel(table_name = nft_metadata_crawler_uris)]
+pub struct NFTMetadataCrawlerURIsQuery {
+    pub token_uri: String,
+    pub raw_image_uri: Option<String>,
+    pub raw_animation_uri: Option<String>,
+    pub cdn_json_uri: Option<String>,
+    pub cdn_image_uri: Option<String>,
+    pub cdn_animation_uri: Option<String>,
+    pub json_parser_retry_count: i32,
+    pub image_optimizer_retry_count: i32,
+    pub animation_optimizer_retry_count: i32,
+    pub last_updated: chrono::NaiveDateTime,
+}
+
+impl NFTMetadataCrawlerURIsQuery {
+    pub fn get_by_token_uri(
+        token_uri: String,
+        conn: &mut PooledConnection<ConnectionManager<PgConnection>>,
+    ) -> diesel::QueryResult<Option<Self>> {
+        nft_metadata_crawler_uris::table
+            .find(token_uri)
+            .first::<NFTMetadataCrawlerURIsQuery>(conn)
+            .optional()
+    }
+
+    pub fn get_by_raw_image_uri(
+        raw_image_uri: Option<String>,
+        conn: &mut PooledConnection<ConnectionManager<PgConnection>>,
+    ) -> diesel::QueryResult<Option<Self>> {
+        if raw_image_uri.is_none() {
```

Comment:
why do you allow passing optional? better to just enforce that it needs to be non null and in the main thread return early or something.

Reply:
The `raw_image_uri` field is an `Option<String>` in the model, and the JSON parsing function also returns an `Option<String>` for it. Makes things flow better in the main thread IMO I think? so I can just do something like:
```
let raw_image_uri: Option<String>, raw_animation_uri: Option<String>  = parseJSON()

model.set_raw_image_uri(raw_image_uri)
model.set_raw_animation_uri(raw_animation_uri)

if force_flag || get_by_raw_image_uri(raw_image_uri) {
  ... do image optimization stuff
}

if force_flag || get_by_raw_animation_uri(raw_animation_uri) {
  ... do animation optimization stuff
}
```

Reply:
But you can just return early before this function right? IMO this is not a great pattern b/c then you have to explain that if it's null it's an no-op.